src/plugins/management/public/management_sections_service.ts

```diff
@@ -17,22 +17,47 @@
  * under the License.
  */
 
-import { ReactElement } from 'react';
 import { ManagementSection, RegisterManagementSectionArgs } from './utils';
-import { managementSections } from './components/management_sections';
+import {
+  IngestSection,
+  DataSection,
+  InsightsAndAlertingSection,
+  SecuritySection,
+  KibanaSection,
+  StackSection,
+} from './components/management_sections';
 
 import {
   ManagementSectionId,
   SectionsServiceSetup,
-  SectionsServiceStart,
   SectionsServiceStartDeps,
+  DefinedSections,
+  ManagementSectionsStartPrivate,
 } from './types';
+import { createGetterSetter } from '../../kibana_utils/public';
+
+const [getSectionsServiceStartPrivate, setSectionsServiceStartPrivate] = createGetterSetter<
+  ManagementSectionsStartPrivate
+>('SectionsServiceStartPrivate');
+
+export { getSectionsServiceStartPrivate };
 
 export class ManagementSectionsService {
-  private sections: Map<ManagementSectionId | string, ManagementSection> = new Map();
+  definedSections: DefinedSections;
```

Comment:
//Major nit

why not just `sections`? (both for the variable and the class)

Reply:
its not necessarily the full set of sections, its just the set of sections that have been defined within this plugin. Sections could be defined and exposed via other plugins.

Reply:
I would then add a comment at least.